test/AutoRest.TestServer.Tests/Infrastructure/CadlRanchNonAzureServer.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Diagnostics;
+using System.IO;
+
+namespace AutoRest.TestServer.Tests.Infrastructure
+{
+    public class CadlRanchNonAzureServer: TestServerBase
+    {
+        public CadlRanchNonAzureServer() : base(GetBaseDirectory(), $"serve {GetScenariosPath()} --port 0 --coverageFile {GetCoverageFilePath()}")
+        {
+        }
+
+        internal static string GetBaseDirectory()
+        {
+            var nodeModules = GetNodeModulesDirectory();
+            return Path.Combine(nodeModules, "@azure-tools", "cadl-ranch");
+        }
+
+        internal static string GetSpecDirectory()
+        {
+            var nodeModules = GetNodeModulesDirectory();
+            return Path.Combine(nodeModules, "@azure-tools", "cadl-ranch-specs");
+        }
+
+        internal static string GetScenariosPath()
+        {
+            return Path.Combine(GetSpecDirectory(), "http");
+        }
+        internal static string GetCoverageFilePath()
+        {
+            return Path.Combine(GetCoverageDirectory(), "cadl-ranch-coverage-csharp-nonAzure.json");
+        }
+
+        protected override void Stop(Process process)
+        {
+            Process.Start(new ProcessStartInfo("node", $"{Path.Combine(GetNodeModulesDirectory(), "@azure-tools", "cadl-ranch", "dist", "cli", "cli.js")} server stop --port {Port}"));
```

Comment:
should we wait for this to exit before we exit?

Reply:
Wait for what? We don't have any information to get.

Reply:
WaitForExit is a blocked call. Do you want an unblocked one?

Reply:
We have two processes in the test run.
1. Process 1: the cadl ranch server process, which starts at the beginning.
2. Process 2: the cadl ranch server stop process, which starts when we exit the test run and it will send a signal to stop Process 1.

Line 38 is starting the Process 2 and line 39 is waiting for Process 1 to exit. So.
1. Process 1 is exactly following what you proposed above.
2. Process 2 we don't have a waitforexit. I don't think we need this because our goal is to just stop Process 1.